### PR TITLE
Snowfakery 2.2 support in CCI

### PR DESCRIPTION
**Pull request by prescod**
# Changes

[Snowfakery 2.2](https://github.com/SFDO-Tooling/Snowfakery/releases/tag/v2.2) is now included with CumulusCI. New features include unique IDs and Numeric Counters.

### CODE REVIEW

File: cumulusci/tasks/bulkdata/generate_from_yaml.py [hunk 1]
```diff
@@ -27,6 +27,9 @@ class GenerateDataFromYaml(BaseGenerateDataTask):
         "vars": {
             "description": "Pass values to override options in the format VAR1:foo,VAR2:bar"
         },
+        "plugin_options": {
+            "description": "Pass values to override plugin options in the format VAR1:foo,VAR2:bar"
```
Comment (davisagli) [inline cumulusci/tasks/bulkdata/generate_from_yaml.py:31]: Don't you need to run this through process_list_of_pairs_dict_arg in order to support that format on the CLI in addition to mappings in yaml?

File: cumulusci/tasks/bulkdata/snowfakery.py [hunk 2]
```diff
@@ -503,6 +507,7 @@ def data_generator_opts(self, working_dir, *args, **kwargs):
             "continuation_file": wd.continuation_file,
             "num_records_tablename": self.run_until.sobject_name or COUNT_REPS,
             "vars": self.recipe_options,
+            "plugin_options": plugin_options,
```
Comment (davisagli) [inline cumulusci/tasks/bulkdata/snowfakery.py:510]: Just wanted to double check that it's intentional this is configurable in the old task but hardcoded in the new one.

File: cumulusci/tasks/bulkdata/generate_from_yaml.py (hunk 1 above)
Comment (prescod) [reply cumulusci/tasks/bulkdata/generate_from_yaml.py:31]: Good point. I was only thinking of this class as an implementation detail of the other one, but of course it is its own task that people can and do use. Fixed in 5e7182fa2

File: cumulusci/tasks/bulkdata/snowfakery.py (hunk 2 above)
Comment (prescod) [reply cumulusci/tasks/bulkdata/snowfakery.py:510]: It's configurable in the old task as an implementation detail of making it available to the new one. Making it configurable here is a reasonable feature request and I can do that before the next CCI release but probably shouldn't scope creep today.

**Review by davisagli — APPROVED** [review]

**Pull request merged**